### ISSUE TITLE
Charging Stats broken / plugin not found: panodata-map-panel

### ISSUE THREAD

**Issue by chicagoandy** [matt-FFFFFF/hassio-addon-teslamate#74]
Charging Stats Panel is broken, it is missing Panodata-map-panel.
![image](https://user-images.githubusercontent.com/81330040/186913867-e9d2efa5-12a2-4e65-88d6-3e6371819e0d.png)

Using TeslaMate .18.1 and Grafana 7.6.0

I have tried added Panodata-map-panel to my Grafana Configuration 

plugins:
  - natel-discrete-panel
  - pr0ps-trackmap-panel
  - grafana-piechart-panel
  - natel-plotly-panel
  - panodata-map-panel

I get the following Grafana Log error:

✔ Downloaded natel-discrete-panel v0.1.1 zip successfully
Please restart Grafana after installing plugins. Refer to Grafana documentation for instructions if necessary.
✔ Downloaded pr0ps-trackmap-panel v2.1.2 zip successfully
Please restart Grafana after installing plugins. Refer to Grafana documentation for instructions if necessary.
✔ Downloaded grafana-piechart-panel v1.6.2 zip successfully
Please restart Grafana after installing plugins. Refer to Grafana documentation for instructions if necessary.
✔ Downloaded natel-plotly-panel v0.0.7 zip successfully
Please restart Grafana after installing plugins. Refer to Grafana documentation for instructions if necessary.
Error: ✗ Plugin not found (Grafana v8.5.5 linux-arm64)
[09:26:51] FATAL: Failed installing Grafana plugin: panodata-map-panel

How do I fix the charging stats dashboard?

**Comment by matt-FFFFFF** [comment]
Hi,

That addon is not in the official Grafana repo. It is not possible to add custom repos to the Grafana Home Assistant Addon maintained by Frenk.

To fix this somebody would have to create a custom Grafana addon or integrate it within the Teslamate one.

**Comment by chicagoandy** [comment]
Ok, can you suggest an alternate way to fix it?

**Comment by mat1990dj** [comment]
+1 with the same issue. It's amazing to have teslamate inside HA but some default dashboards are using that panodata-map-panel and we're missing that amazing map

**Comment by ThePatricide** [comment]
I really hope this can be fixed one day, although it might never come. Would love to analyse the charging curve when going abroad. Now not possible without the natel-plotly-panel.

**Comment by llluis** [comment]
I forked the community addon to allow the installation of unsigned plugins directly from a URL, based on the previous solution by Matt. This solves the issue for panodata.
I'll try to submit a PR to see if gets accepted like this by Frenk.

![image](https://github.com/matt-FFFFFF/hassio-addon-teslamate/assets/708868/f9a55bda-1b73-4bda-b400-a8aab3cc9244)

If you want to try it, you can add it from my repo: https://github.com/llluis/hassio-addon-grafana
You will have to change your Grafana host in TeslaMate addon configuration to ```8fba5054-grafana```

In the plugins section of Grafana, use:
```
plugins:
  - name: natel-discrete-panel
  - name: natel-plotly-panel
  - name: pr0ps-trackmap-panel
  - name: grafana-piechart-panel
  - name: panodata-map-panel
    url: https://github.com/panodata/panodata-map-panel/releases/download/0.16.0/panodata-map-panel-0.16.0.zip
```

And the environment variable:
```
env_vars:
  - name: GF_PLUGINS_ALLOW_LOADING_UNSIGNED_PLUGINS
    value: panodata-map-panel
```

Remember to restart the TeslaMate addon so the dashboards are installed in the new Grafana instance.

**Comment by ohadlevy** [comment]
@llluis I've tried to install it and got the following error:
I've also noted the lack of version (only dev)

```
The command '/bin/bash -o pipefail -c ARCH="${BUILD_ARCH}" && if [ "${BUILD_ARCH}" = "aarch64" ]; then ARCH="arm64"; fi && if [ "${BUILD_ARCH}" = "armv7" ]; then ARCH="armhf"; fi && curl -J -L -o /tmp/grafana.deb "https://dl.grafana.com/oss/release/grafana_${GRAFANA_VERSION#v}_${ARCH}.deb" && apt-get update && apt-get install -y --no-install-recommends libfontconfig1=2.13.1-4.2 fonts-noto-cjk=1:20201206-cjk+repack1-1 memcached=1.6.9+dfsg-1 nginx=1.18.0-6.1+deb11u3 && dpkg --install /tmp/grafana.deb && if [ "${BUILD_ARCH}" = "amd64" ]; then apt-get install -y --no-install-recommends libasound2=1.2.4-1.1 libcups2=2.3.3op2-3+deb11u2 libdbus-1-3=1.12.24-0+deb11u1 libdrm-amdgpu1=2.4.104-1 libgbm1=20.3.5-1 libglib2.0-0=2.66.8-1 libgtk-3-0=3.24.24-4+deb11u3 libnss3=2:3.61-1+deb11u3 libx11-6=2:1.7.2-1 libx11-xcb1=2:1.7.2-1 libxcb-dri3-0=1.14-3 libxcomposite1=1:0.4.5-1 libxcursor1=1:1.2.0-2 libxdamage1=1:1.1.5-2 libxext6=2:1.3.3-1.1 libxfixes3=1:5.0.3-2 libxi6=2:1.7.10-1 libxrandr2=2:1.5.1-1 libxrender1=1:0.9.10-1 libxshmfence1=1.3-1 libxss1=1:1.2.3-1 libxtst6=2:1.2.3-1 && grafana-cli plugins install "grafana-image-renderer" "${GRAFANA_IMAGE_RENDERER_VERSION#v}"; fi && rm -fr /tmp/* /var/{cache,log}/* /var/lib/apt/lists/* /etc/nginx && mkdir -p /var/log/nginx && touch /var/log/nginx/error.log' returned a non-zero code: 159
```

**Comment by ohadlevy** [comment]
I assume its mismatch in arch, as the following appears in the logfile:

```
hassio_supervisor[641]:  ---> [Warning] The requested image's platform (linux/arm64) does not match the detected host platform (linux/arm/v8) and no specific platform was requested
```

**Comment by llluis** [comment]
> @llluis I've tried to install it and got the following error: I've also noted the lack of version (only dev)

Does it work with the official add-on? 
I didn't change anything to cause such error, so I would imagine it would also not work with it. 

It's showing dev because I forked main and not the 8.3.0 tag from the original repo, but they are identical:

![image](https://github.com/matt-FFFFFF/hassio-addon-teslamate/assets/708868/1b67dbda-32d5-4d38-a632-73433c4cc8d2)

Finally, I changed the configuration after some comments / requests from frenck. So, if you pull the default branch from my repo (dev), configuration is a bit different. See: https://github.com/hassio-addons/addon-grafana/blob/9a7244fe9ab65f32c127a7bd6eea8a10c82a229a/grafana/DOCS.md#option-custom_plugins

(it's a no breaking change now, you keep your usual plugins and add a new custom_plugins for panodata)

```
plugins:
  - natel-discrete-panel
  - natel-plotly-panel
  - pr0ps-trackmap-panel
  - grafana-piechart-panel
custom_plugins:
  - name: panodata-map-panel
    url: https://github.com/panodata/panodata-map-panel/releases/download/0.16.0/panodata-map-panel-0.16.0.zip
    unsigned: true
```

Lets hope he approves it this time.
